tools/run.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+# Copyright: (c) 2020, Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import argparse
+import json
+import os
+import random
+import subprocess
+import sys
+
+
+CONTAINER = 'quay.io/ansible/default-test-container:1.12'
+
+COLORS = {
+    'emph': 1,
+    'gray': 37,
+    'black': 30,
+    'white': 97,
+    'green': 32,
+    'red': 31,
+    'yellow': 33,
+}
+
+
+def colorize(text, color, use_color=True):
+    if not use_color:
+        return text
+    color_id = COLORS.get(color)
+    if color_id is None:
+        return text
+    return '\x1b[{0}m{1}\x1b[0m'.format(color_id, text)
+
+
+def run(command, catch_output=False, use_color=True):
+    sys.stdout.write(colorize('[RUN] {0}\n'.format(' '.join(command)), 'emph', use_color))
+    sys.stdout.flush()
+    if catch_output:
+        p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        return p.returncode, stdout, stderr
+    else:
+        return subprocess.call(command), None, None
+
+
+def get_common_parent(*directories):
+    parent = directories[0]
+    for dir in directories[1:]:
+        while os.path.relpath(dir, parent).startswith('..'):
+            old_parent, parent = parent, os.path.dirname(parent)
+            if old_parent == parent:
+                break
+    return parent
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Extra sanity test runner.')
+    parser.add_argument('--color',
+                        action='store_true',
+                        help='use ANSI colors')
+    parser.add_argument('targets',
+                        metavar='TARGET',
+                        nargs='*',
+                        help='targets')
+
+    args = parser.parse_args()
+
+    use_color = sys.stdout.isatty()
+    if args.color:
+        use_color = True
+
+    cwd = os.getcwd()
+    root = cwd
+    my_dir = '.'
+    try:
+        my_dir = os.path.dirname(__file__)
+        root = get_common_parent(root, my_dir)
+    except Exception as e:
```

Comment:
```suggestion
    except Exception:
```

Reply:
Will fix that. I'll also let pylint and flake8 run over this (I think I already looked at what flake8 says) to remove potential linter warnings.